client/lib/support/support-user/localforage-bypass.js

```diff
@@ -0,0 +1,230 @@
+/**
+ * External dependencies
+ */
+import debugModule from 'debug';
+
+const debug = debugModule( 'calypso:support-user' );
+
+// This module defines a custom localForage driver which bypasses all persistent
+// storage. Any calls to read/write data using localForage instead access a temporary
+// in-memory store which is lost on page reload. This driver is used to sandbox
+// a user's data while support-user is active, ensuring it does not contaminate the
+// original user/vice versa.
+// Copied with modifications from localForage source: localforage/test/dummyStorageDriver.js
```

Comment:
A small thing, but we may want to point to [`CREDITS.md`](https://github.com/Automattic/wp-calypso/blob/master/CREDITS.md#httpsgithubcommozillalocalforage) for licensing information.